src/Neo/SmartContract/Native/ContractMethodAttribute.cs

```diff
@@ -23,12 +23,19 @@ internal class ContractMethodAttribute : Attribute
         public long CpuFee { get; init; }
         public long StorageFee { get; init; }
         public Hardfork? ActiveIn { get; init; } = null;
+        public Hardfork? DeprecatedIn { get; init; } = null;
 
         public ContractMethodAttribute() { }
 
         public ContractMethodAttribute(Hardfork activeIn)
         {
             ActiveIn = activeIn;
         }
+
+        public ContractMethodAttribute(bool isDeprecated, Hardfork deprecatedIn)
```

Comment:
```suggestion
        public ContractMethodAttribute(Hardfork? activeIn, Hardfork deprecatedIn)
```
I'd suggest to use this option, and just put `null` as the first argument for those methods that were always active but need to be deprecated.

Reply:
Existing solution already works, i may update it after we have confirmed that this pr works fine.

Reply:
Ref. https://github.com/neo-project/neo/issues/3210#issuecomment-2114292377 as TODO.